include/cuda/std/detail/libcxx/include/atomic

```diff
@@ -1239,7 +1239,7 @@ _LIBCUDACXX_INLINE_VISIBILITY void __cxx_atomic_wait(__cxx_atomic_impl<_Tp, _Sco
 }
 
 // general atomic<T>/atomic_ref<T>
-template <class _Tp, int _Sco = 0, bool = is_integral<_Tp>::value && !is_same<_Tp, bool>::value>
+template <class _Tp, int _Sco = 0, bool = (is_integral<_Tp>::value || is_floating_point<_Tp>::value) && !is_same<_Tp, bool>::value>
```

Comment:
Because we are enabling the arithmetic operators in `cuda::std::atomics` as well we would need to extend the tests to exercise floating point.

These tests live in `tests/std/atomics/` rather than `tests/cuda/atomics`

Reply:
.upstream-tests/test/std/atomics/atomics.types.operations/atomics.types.operations.arith is this the right directory?

Reply:
Ah, I figure I need to add some more tests under the `std/atomics` directory. Basically mirror every test over integral types.

Reply:
Yeah, the integral tests would be a good start. Replicating the tests and doing the dispatch yourself for float/double would probably be sufficient. I don't think figuring out how to fit bitwise/arithmetic dispatches into the atomics_helpers there would be worth the effort.